session/txn.go

```diff
@@ -447,11 +453,13 @@ func keyNeedToLock(k, v []byte) bool {
 		// meta key always need to lock.
 		return true
 	}
-	isDelete := len(v) == 0
-	if isDelete {
-		// only need to delete row key.
-		return k[10] == 'r'
+
+	// do not lock row key for delete operation,
```

Comment:
Why not?

Reply:
Maybe another transaction is updating a non-index column, we still need to lock the row key.